src/app/modules/pictureapp-explore/components/photo-uploader-dialog/components/photo-uploader-addphoto-firststep/photo-uploader-addphoto-firststep.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit, Input, EventEmitter, Output } from '@angular/core';
+import { FileUploader } from 'ng2-file-upload';
+import { environment } from 'src/environments/environment';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { PhotoUploaderDialogComponent } from '../../photo-uploader-dialog.component';
+import { PhotoUploaderModel } from 'src/app/models/photo-uploader-model';
+
+@Component({
+  selector: 'app-photo-uploader-addphoto-firststep',
+  templateUrl: './photo-uploader-addphoto-firststep.component.html',
+  styleUrls: ['./photo-uploader-addphoto-firststep.component.scss']
+})
+export class PhotoUploaderAddphotoFirststepComponent implements OnInit {
+
+  @Input() fileInput: File;
+
+  public uploader: FileUploader;
+  public filePreviewPaths: SafeUrl[];
+
+  public hasBaseDropZoneOver = false;
+  baseUrl = environment.apiUrl;
+  constructor(private sanitizer: DomSanitizer) { }
+
+  ngOnInit() {
+    this.initUploader();
+    this.setCurrentPhotoForUploader();
+    this.prepareImagePreviewForDropedImages();
+  }
+
+  initUploader() {
+    this.uploader = new FileUploader({
```

Comment:
Why do we use different upload approaches? Here is a FileUploader whilst couple files above there is HttpClient?